utils/dune.ts

```diff
@@ -0,0 +1,27 @@
+import get from 'lodash/get';
+
+import { PREDICTION_DAA_QUERY_ID } from '../constants';
+
+const duneApiCall = async ({ queryId }: { queryId: string }) => {
+  try {
+    const response = await fetch(`https://api.dune.com/api/v1/query/${queryId}/results`, {
+      headers: {
+        'X-Dune-API-Key': process.env.NEXT_PUBLIC_DUNE_API_KEY as string,
+      },
+    });
+    const json = await response.json();
+    return json;
+  } catch (error) {
+    console.error(error);
+  }
+
+  return null;
+};
+
+export const getPredictionDaa = async () => {
+  const json = await duneApiCall({
+    queryId: PREDICTION_DAA_QUERY_ID,
+  });
+  const average = get(json, 'result.rows[0].avg_7d_active_agents_count');
+  return Math.ceil(average);
+};
```

Comment:
error is caught in `duneApiCall` but then an invalid value `null` is passed as the result from duneApiCall

this will cause an error in `getPredictionDaa`, as it expects json in a particular structure

avoid null response here, `null` should be used with intention when you _want_ something to have no value. it's not an "erroneous" value, unlike `undefined`. it conceals that there is a problem.

i would either:

- remove the try/catch completely and catch the error after the promise resolves, i.e. where it's used, for example in `getPredictionDaa`

- or, create a default fallback that won't break `result.rows[0].avg_7d_active_agents_count` -- `json` as `null` certainly won't work here, and will result in the `get` function returning `undefined` to the `Math.ceil`


```suggestion
const duneApiCall = async ({ queryId }: { queryId: string }) => {
    const response = await fetch(`https://api.dune.com/api/v1/query/${queryId}/results`, {
      headers: {
        'X-Dune-API-Key': process.env.NEXT_PUBLIC_DUNE_API_KEY as string,
      },
    });
    const json = await response.json();
    return json;
};

export const getPredictionDaa = async () => {
  const json = await duneApiCall({
    queryId: PREDICTION_DAA_QUERY_ID,
  });
  const average = get(json, 'result.rows[0].avg_7d_active_agents_count', 0); // 0 fallback, or allow error to propogate to the component that calls this function
  return Math.ceil(average);
};
```

Reply:
updated, lmkyt @truemiller 🙏